src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2101,6 +2101,16 @@ SelectsFromDistributedTable(List *rangeTableList, Query *query)
 			continue;
```

Comment:
semi-related to this PR: I'm surprised to see (a) hard errors in router planner that we can actually run via recursive planner (b) `SelectsFromDistributedTable ()` is called in  `SingleShardTaskList` not in `ModifyQuerySupported()`